### ISSUE TITLE
Support multiple icons/tooltips in workbench toolbar based on context/when clause

### ISSUE THREAD

**Issue by Yoyokrazy** [microsoft/vscode#187097]
TLDR: unify multiple distinct actions into a single action, with different icons/tooltips/functions based on notebook state

## Details:
Currently in the notebook toolbar there is an action `Go to` that changes depending on the state of the notebook. Currently we have the following actions:
- Go to Running cell
https://github.com/microsoft/vscode/blob/0aaacf4b1206f02b081296f49a14c3b3de45cb29/src/vs/workbench/contrib/notebook/browser/controller/executeActions.ts#L613-L654

- Go to Most Recently Failed Cell
https://github.com/microsoft/vscode/blob/0aaacf4b1206f02b081296f49a14c3b3de45cb29/src/vs/workbench/contrib/notebook/browser/controller/executeActions.ts#L613-L654

Alongside these, there is a plan to add `Go to Recently Run Cell` with #185902. Before that can be merged, there's an issue with hiding these actions. With the current state of the notebook toolbar, if a user hides the `Go to Recently Failed Cell` action, they will still see the `go to currently running cell` action. 

In the future, because `Go To Recently Run Cell` will be visible when the other two GoTo options aren't valid, the user should be able to hide the entire "goto suite" at the same time. Currently I don't believe that it's possible to unify these three actions into a single one.

cc: @jrieken // @sbatten // @rebornix

**Comment by jrieken** [comment]
> TLDR: unify multiple distinct actions into a single action, with different icons/tooltips/functions based on notebook state

Should this be understood as submenu which isn't a menu but a stack of actions where context keys define which action is on top?

**Comment by rebornix** [comment]
> Should this be understood as submenu which isn't a menu but a stack of actions where context keys define which action is on top?

I like this idea, it's worth a try.

**Comment by Yoyokrazy** [comment]
Submenu for each action, with topmost action being populated as the rendered action. Override the standard submenu view to not render upon click, but rather execute the rendered or "topmost" action.
